lib/client/api_matrix.es6.js

```diff
@@ -125,31 +274,125 @@ foam.CLASS({
+        CSV format.`,
+      args: [
+        {
+          name: 'browserKeys',
+          typeName: 'StringArray',
+          documentation: `A string array contains valid browser keys.`,
+        },
+        {
+          name: 'matrix',
+          typeName: 'JSON',
+          documentation: `The JSON object returned from toMatrix.`,
+        },
+      ],
+      returns: {
+        typeName: 'String',
+        documentation: `Returns a string of csv format.`,
+      },
+      code: function(browserKeys, matrix) {
+        let table = [];
+        // Add table header.
+        table.push(['Interface', 'API'].concat(browserKeys));
+        let interfaces = Object.keys(matrix).sort();
+        for (let i = 0; i < interfaces.length; i++) {
+          let interfaceName = interfaces[i];
+          let APIs = Object.keys(matrix[interfaceName]).sort();
+          for (let j = 0; j < APIs.length; j++) {
+            let apiName = APIs[j];
+            let row = [interfaceName, apiName];
+            for (let k = 0; k < browserKeys.length; k++) {
+              row.push(matrix[interfaceName][apiName]
+                .hasOwnProperty(browserKeys[k]));
             }
+            table.push(row);
           }
-          let csv = '';
-          for (let i = 0; i < table.length; i++) {
-            csv += table[i].join(',');
-            csv += '\n';
-          }
-          return csv;
-        });
+        }
+        let csv = '';
+        for (let i = 0; i < table.length; i++) {
+          csv += table[i].join(',');
+          csv += '\n';
+        }
+        return csv;
+      },
+    },
+    {
+      name: 'getAnalytics',
+      documentation: `Takes an array of browser keys, produces a JSON
+        contains number of proprietary APIs and falling behind APIs for
+        each selected browsers.`,
+      args: [
+        {
+          name: 'browserKeys',
+          typeName: 'StringArray',
+          documentation: `A string array contains valid browser keys.`,
+        },
+      ],
+      returns: {
+        typeName: 'JSON',
+        documentation: `A JSON of the form:
+          {
+            browserKey:
+            {
+              total: Int,
+              proprietary: Int,
+              fallBehind: Int
+            }
+          }.`,
+      },
+      code: function(browserKeys) {
+        return this.browserApiDAO
+          .where(this.mlang
+            .IN(this.BrowserWebInterfaceJunction.SOURCE_ID, browserKeys))
+          .select(this.mlang.GROUP_BY(
+            this.BrowserWebInterfaceJunction.TARGET_ID,
+            this.ArraySink.create()))
+          .then((groups) => {
+            let result = {};
+            let totalBrowsers = browserKeys.length;
+            let interfaceKeys = groups.groupKeys;
+            for (let i = 0; i < browserKeys.length; i++) {
+              result[browserKeys[i]] = {
+                // Total is the total number of APIs belongs to
```

Comment:
Should the documentation for the fields in the returned structure be moved up into the function's documentation?